File: library/src/main/java/com/github/kongpf8848/rxhttp/util/TypeUtil.java
package com.github.kongpf8848.rxhttp.util;

import java.lang.reflect.ParameterizedType;
import java.lang.reflect.Type;

public class TypeUtil {

    //获取参数类型
    public static Type getType(Class<?> subclass) {
        Type superclass = subclass.getGenericSuperclass();
        if (superclass instanceof Class) {
            return subclass;
        } else {
            ParameterizedType parameterized = (ParameterizedType) superclass;
            return parameterized.getActualTypeArguments()[0];
        }
    }
}

File: RxHttp/src/main/java/com/github/kongpf8848/rxhttp/util/TypeUtil.kt
package com.github.kongpf8848.rxhttp.util

import java.lang.reflect.ParameterizedType
import java.lang.reflect.Type

object TypeUtil {
    //获取参数类型
    fun getType(subclass: Class<*>): Type {
        val superclass = subclass.genericSuperclass
        return if (superclass is Class<*>) {
            subclass
        } else {
            val parameterized = superclass as ParameterizedType
            parameterized.actualTypeArguments[0]
        }
    }
}